Build error on mac osx (Julia 0.4 latest)

Hi all, 

I am trying to get the latest julia build for my mac but I can't seem to compile it. Near the end of the compilation I get the following errors
```
[ 22%] Building C object CMakeFiles/git2.dir/src/unix/map.c.o
[ 22%] Building C object CMakeFiles/git2.dir/src/unix/realpath.c.o
[ 22%] Building C object CMakeFiles/git2.dir/deps/http-parser/http_parser.c.o
Linking C shared library libgit2.dylib
Undefined symbols for architecture x86_64:
  "_libiconv", referenced from:
      _git_path_iconv in path.c.o
  "_libiconv_close", referenced from:
      _git_path_direach in path.c.o
      _git_path_iconv_clear in path.c.o
      _git_path_diriter_free in path.c.o
      _git_path_dirload in path.c.o
  "_libiconv_open", referenced from:
      _git_path_direach in path.c.o
      _git_path_iconv_init_precompose in path.c.o
      _git_path_diriter_init in path.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [libgit2.0.23.1.dylib] Error 1
make[3]: *** [CMakeFiles/git2.dir/all] Error 2
make[2]: *** [all] Error 2
make[1]: *** [libgit2/build/libgit2.dylib] Error 2
make: *** [julia-deps] Error 2
```
Looks like a linker error. But am not sure what is causing this. I am just running the default make command. I am on OSX Yosemite.

Any clues?

thanks,

Also https://github.com/libgit2/libgit2/issues/3296 - a workaround might be to temporarily take fink off your path, or maybe just fink's pkg-config.



This wasn't the error you had, but I suspect someone else is going to hit it and get confused. If anyone reading this hits an error that looks like
```
error: Your local changes to the following files would be overwritten by checkout:
    CMakeLists.txt
    src/openssl_stream.c
Please, commit your changes or stash them before you can switch branches.
Aborting
make[1]: *** [libgit2/CMakeLists.txt] Error 1
make: *** [julia-deps] Error 2
```
then you just need to run `make -C deps distclean-libgit2` and you should be all set. That won't quite help @GravityAssisted until he fiddles with his fink / pkg-config path though.

--- a/Make.inc
+++ b/Make.inc
@@ -160,6 +160,9 @@ BUILD_LLDB = 0
 # Path to cmake (override in Make.user if needed)
 CMAKE ?= cmake
 
+# Point pkg-config to only look at our libraries
+export PKG_CONFIG_LIBDIR = $(JULIAHOME)/usr/lib/pkgconfig
+
 # Figure out OS and architecture
 BUILD_OS := $(shell uname)
 

